src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -333,6 +333,11 @@ private HttpMessageHandler SetupHandlerChain()
         protected internal override Task<HttpResponseMessage> SendAsync(
             HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request), SR.net_http_handler_norequest);
```

Comment:
`request` is already checked for `null` in `HttpClient` prior to calling this; a `Debug.Assert` might be more appropriate here.